### PR TITLE
Assignment7 - Chester Southwood

### PR DESCRIPTION
**Pull request by Chester-Southwood**
I had trouble uploading Tuesday night due to Visual Studios acting up. I tried updating, but the update essentially reinstalled visual studios. By the time I tried to upload before midnight, the icon that usually says "pull request" said "View 120". I tried going around but wasn't able to find a way until the next morning when manually having to make request by forks. Before I was able to do a pull request I looked online and one user recommended cherry picking instead of creating branch. I did so, but I think that added more on my branch then I needed; not solving the problem. The actual homework will be submitted on time.

### CODE REVIEW

File: SecretSanta/SecretSanta.Web/Controllers/UsersController.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        public async Task<IActionResult> Index()
+        {
+            using (var httpClient = ClientFactory.CreateClient("SecretSantaApi"))
+            {
+                var response = await httpClient.GetAsync("/api/users");
+                if (response.IsSuccessStatusCode)
+                {
+                    var content = await response.Content.ReadAsStringAsync();
+
+                    ViewBag.Users = JsonConvert.DeserializeObject<ICollection<UserViewModel>>(content);
```
Comment (COsborn2) [inline SecretSanta/SecretSanta.Web/Controllers/UsersController.cs]: If you look at the updated Lecture-2019.02.21 Stokes added the NSwagStudio way of doing things, which I think is required for this assignment. If you pull in that Lecture and start from there its actually easier than this way of doing things!

File: SecretSanta/SecretSanta.Web/Views/Users/Add.cshtml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿@model UserInputViewModel
+<form asp-route="Add">
+    <div class="field">
+        <label class="label" asp-for="FirstName">First Name</label>
+        <div class="control">
+            <input class="input" type="text" asp-for="FirstName" />
```
Comment (COsborn2) [inline SecretSanta/SecretSanta.Web/Views/Users/Add.cshtml]: Dont forget to add a `<span>` here to enforce FirstName validation not being null.

File: SecretSanta/SecretSanta.Web/Views/Users/Index.cshtml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿@*
+    For more information on enabling MVC for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
```
Comment (COsborn2) [inline SecretSanta/SecretSanta.Web/Views/Users/Index.cshtml]: I think you can delete this boilerplate stuff.

**Comment by breaman** [comment]
Overall nice work on the assignment. Hit the points we needed to hit. Well done.

**Pull request closed without merging**